app/controllers/charges_controller.rb

```diff
@@ -0,0 +1,243 @@
+      end
+    end
+
+    # Check if any time slot is selected both for single booking and multibooking
+    def check_time_slot
+      if (params[:user].nil? || params[:user][:temp_availability].nil?)
+        flash[:alert] = "Please choose a time slot."
+        if params[:action] == "checkout_multiple"
+          redirect_to :controller => "user", :action => "multiple_booking" 
+        else
+          redirect_to :controller => "user", :action => "booking" 
+        end
+        return
+      end
+    end
+
+    def create_stripe_customer(email, stripe_token)
+      customer = Stripe::Customer.create({
+        email: email,
+        source: stripe_token,
+      })
+      return customer
+
+    end
+
+    def create_stripe_charges(amount, customer)
+      charge = Stripe::Charge.create({
+        customer: customer.id,
+        amount: amount,
+        description: 'Rails Stripe customer',
+        currency: 'usd',
+      })
+      return charge
+    end
+
+    def create_invoice(amount, customer)
+      Stripe::InvoiceItem.create({
+          amount: amount,
+          currency: 'usd',
+          customer: customer.id,
+          description: 'Lesson fee',
+      })
+      invoice = Stripe::Invoice.create({
+        customer: customer.id,
+        billing: 'send_invoice',
+        days_until_due: 30,
+      })
+
+      return invoice
+
+    end
+
+    # Checking if the selected event time is before the current time
+    def check_time_past(event_start, multiple_booking, year)
+      if year > DateTime.now.year.to_i
+        return false
+      end
+      if event_start.past?
+        flash[:alert] = "Please choose a future time slot."
+        if multiple_booking
+          redirect_to :controller => "user", :action => "multiple_booking"
+        else
+          redirect_to :controller => "user", :action => "booking"
+        end
+        return true
+      end
+      return false
+    end
+
+    # Returns a list of [day, month, year] after increasing the day by 7(a week). 
+    def update_day_month(day_index, month_index, year_index)
+       # incrementing by 7 days and updating month and day and year
+       day_index += 7
+       # Checking if it is going to a new year
+       if (day_index > 31) and (month_index == 12)
+        next_year = year_index + 1
+        updated_day_index = day_index - Time.days_in_month(1, year = next_year)
+        return [updated_day_index, 1, next_year]
+       end
+       # If it's not going to a new year, check if it is going to a new month and update accordingly
+       if (day_index > Time.days_in_month(month_index, year = year_index))
+        updated_day_index = day_index - Time.days_in_month(month_index, year = year_index)
+        updated_month_index = month_index + 1
+        return [updated_day_index, updated_month_index, year_index]
+       else
+        return [day_index, month_index, year_index]
+       end
+    end
+
+    public 
+    def new
+    end
+    
+    def create
```

Comment:
This method is a beast. Please refactor - consider breaking up into more methods if needed

Reply:
Ok we will try, though this we honestly already kinda cracked our heads trying to make it simpler, but still couldn’t. We will try to refactor a bit more

Reply:
I'm sure that there's some logical breaks where you can extract and call it something different. Also, most of the work should be done in a model, not controller. So if anything else, break it up and move the pieces to model